cmd/cluster-agent/api/v1/languagedetection/handler.go

```diff
@@ -0,0 +1,116 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package languagedetection
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	pbgo "github.com/DataDog/datadog-agent/pkg/proto/pbgo/process"
+)
+
+var (
+	// statusSuccess is the value for the "status" tag that represents a successful operation
+	statusSuccess = "success"
+	// statusError is the value for the "status" tag that represents an error
+	statusError = "error"
+)
+
+type handlerConfig struct {
+	enabled       bool
+	languageTTL   time.Duration
+	cleanupPeriod time.Duration
+}
+
+type languageDetectionHandler struct {
+	ownersLanguages OwnersLanguages
```

Comment:
```suggestion
	ownersLanguages *OwnersLanguages
```
I think a pointer could be useful here.